docs/ai/javascript.rst

```diff
@@ -0,0 +1,167 @@
+.. _ref_ai_javascript:
+
+==========
+JavaScript
+==========
+
+:edb-alt-title: EdgeDB AI's JavaScript package
+
+``@edgedb/ai`` offers a convenient wrapper around ``ext::ai``. Install it with
+npm or via your package manager of choice:
+
+.. code-block:: bash
+
+    $ npm install @edgedb/ai # or
+    $ yarn add @edgedb/ai # or
+    $ pnpm add @edgedb/ai
+
+
+Usage
+=====
+
+Start by importing ``createClient`` from ``edgedb`` and ``createAI`` from
+``@edgedb/ai``:
+
+.. code-block:: typescript
+
+    import { createClient } from "edgedb";
+    import { createAI } from "@edgedb/ai";
+
+Create an EdgeDB client. Create an instance of the AI client by passing in the
+EdgeDB client and any options for the AI provider (like the text generation
+model):
+
+.. code-block:: typescript
+
+    const client = createClient();
+
+    const gpt4Ai = createAI(client, {
+      model: "gpt-4-turbo-preview",
+    });
+
+Add your query as context:
+
+.. code-block:: typescript
+
+    const astronomyAi = gpt4Ai.withContext({
+      query: "Astronomy"
+    });
+
+The default text generation prompt will ask your selected provider to limit
+answer to information provided in the context and will pass the queried
+objects' AI index as context along with that prompt.
+
+Call your AI client's ``queryRag`` method, passing in a text query.
+
+.. code-block:: typescript
+
+    console.time("gpt-4 Time");
+    console.log(
+      await astronomyAi.queryRag("What color is the sky on Mars?")
+    );
+    console.timeEnd("gpt-4 Time");
```

Comment:
Is leaving `console.time` here relevant?

Reply:
I don't think so. I'll clean that up.